bee-network/bee-autopeering/src/peer/peerlist.rs

```diff
@@ -0,0 +1,391 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{peer_id::PeerId, Peer};
+
+use crate::{
+    discovery::manager::VERIFICATION_EXPIRATION_SECS,
+    time::{self, Timestamp},
+};
+
+use serde::{
+    de::{SeqAccess, Visitor},
+    ser::SerializeStruct,
+    Deserialize, Serialize,
+};
+
+use std::{
+    collections::{HashSet, VecDeque},
+    fmt,
+    sync::{Arc, RwLock, RwLockReadGuard, RwLockWriteGuard},
```

Comment:
Same, parking_lot

Reply:
see my other comment: I'd bench and see if it gains us anything before adding another dep. Maybe it's just me, but I don't think performance considerations should necessarily be part of a 0.1.0 release.